classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -488,4 +495,74 @@ protected function setFallbackVersion(): ?string
         }
         return null;
     }
+
+    /**
+     * Check the contexts' contact details before upgrade
+     *
+     * @return void
+     * @throws Exception
+     */
+    protected function checkContactSetting(): void
+    {
+        $primaryLocale = DB::table('site')
+            ->select(['primary_locale'])
+            ->first()
+            ->primary_locale;
+
+        // Get all the enabled context with context name/title
+        $contexts = DB::table("{$this->getContextTable()} AS context")
+            ->select([
+                "context.{$this->getContextKeyField()}", 
+                "context.enabled", 
+                "context_setting.locale",
+                "context_setting.setting_name",
+                "context_setting.setting_value"
+            ])
+            ->join("{$this->getContextSettingsTable()} AS context_setting", fn($join) =>
+                $join
+                    ->on("context.{$this->getContextKeyField()}", "=", "context_setting.{$this->getContextKeyField()}")
+                    ->where("context_setting.locale", "=", $primaryLocale)
+                    ->where("context_setting.setting_name", "=", "name")
+            )
+            ->where("context.enabled", 1)
+            ->get();
+        
+        // Get all the context IDs for which contact name/email is set
+        $contextIdsWithValidContact = DB::table("{$this->getContextTable()} AS context")
+            ->select(["context.{$this->getContextKeyField()}"])
+            ->join("{$this->getContextSettingsTable()} AS contact_name_setting", fn($join) => 
+                $join
+                    ->on("contact_name_setting.{$this->getContextKeyField()}", "=", "context.{$this->getContextKeyField()}")
+                    ->where("contact_name_setting.setting_name", "=", "contactName")
+            )
+            ->leftJoin("{$this->getContextSettingsTable()} AS contact_email_setting", fn($join) => 
+                $join
+                    ->on("contact_email_setting.{$this->getContextKeyField()}", "=", "context.{$this->getContextKeyField()}")
+                    ->where("contact_email_setting.setting_name", "=", "contactEmail")
+            )
+            ->whereNotNull("contact_email_setting.{$this->getContextKeyField()}")
```

Comment:
Can journal ID be null here? Just clarifying for myself this check

Reply:
No and it's no necessary. updated .